src/main/java/net/dv8tion/jda/internal/requests/restaction/MessageActionImpl.java

```diff
@@ -383,8 +383,13 @@ public MessageActionImpl retainFilesById(@Nonnull Collection<String> ids)
     public MessageActionImpl setActionRows(@Nonnull ActionRow... rows)
     {
         Checks.noneNull(rows, "ActionRows");
+
+        if (Arrays.stream(rows).anyMatch(actionRow -> !actionRow.isMessageCompatible()))
+            throw new IllegalArgumentException("Provided ActionRow is not compatible with Messages!");
```

Comment:
Unfortunately, this check is only minimally helpful.
An ActionRow component is, currently, always going to be compatible, so really it is an underlying component inside of the ActionRow that is problematic.

However, this error doesn't tell you:
 - which of the action rows you provided was problematic
 - which of the components in that action row was problematic

An ideal error would be like
`ActionRow[0].components[2] - <TextInput> is not compatible with Messages!`

It tells you specifically _what_ is the problem and _where_ the problem is.

Reply:
This applies to all checks that follow this pattern.